src/elfpy/user.py

```diff
@@ -4,178 +4,114 @@
-        if self.verbose and streak_luck > 0.98:
-            direction_index = 1 - round(sum(self.days_trades) / len(self.days_trades))
-            print(
-                f" days_trades={self.days_trades}+{direction_index}k={sum(self.days_trades)}"
-                f" n={len(self.days_trades)} ratio={sum(self.days_trades)/len(self.days_trades)}"
-                f" streak_luck: {streak_luck}"
-            )
-        else:
-            if 0 < apy_distance_from_mid_when_in_range < 1:
-                actual_convergence_strength = (
-                    actual_convergence_strength + (1 - actual_convergence_strength) * streak_luck**1.5
-                )  # force convergence when on bad streaks
-            direction_index = (
-                convergence_direction if self.rng.random() < actual_convergence_strength else 1 - convergence_direction
-            )
-        self.days_trades.append(direction_index)
-        if self.verbose and pool_apy > 0.2:
-            print(
-                f" days_trades={self.days_trades}"
-                f" k={sum(self.days_trades)}"
-                f" n={len(self.days_trades)}"
-                f" ratio={sum(self.days_trades)/len(self.days_trades)}"
-                f" streak_luck: {streak_luck}"
-            )
-            if self.pool_apy_target_range is not None:
-                print(btest)
-                print(f"expected_proportion={expected_proportion}")
-                print(
-                    f" pool_apy = {pool_apy:,.4%} apy_distance_in_target_range ="
-                    f" {apy_distance_in_target_range},"
-                    " apy_distance_from_mid_when_in_range ="
-                    f" {apy_distance_from_mid_when_in_range},"
-                    " actual_convergence_strength ="
-                    f" {actual_convergence_strength}, direction_index ="
-                    f" {direction_index}"
-                )
-        # Append new values the internal user state
-        self.user_state["direction_index"].append(direction_index)
-        self.user_state["apy_distance_from_mid_when_in_range"].append(apy_distance_from_mid_when_in_range)
-        self.user_state["apy_distance_in_target_range"].append(apy_distance_in_target_range)
-        self.user_state["actual_convergence_strength"].append(actual_convergence_strength)
-        self.user_state["expected_proportion"].append(expected_proportion)
-        self.user_state["streak_luck"].append(streak_luck)
-        self.user_state["btest"].append(btest)
-        return direction_index
+        output_with_fee = self.market.pricing_model.calc_out_given_in(
+            self.wallet["base_in_wallet"],
+            self.market.share_reserves,
+            self.market.bond_reserves,
+            "base",
+            self.market.fee_percent,
+            stretched_time_remaining,
+            self.market.init_share_price,
+            self.market.share_price,
+        )[1]
+        max_short = self.wallet["base_in_wallet"] + output_with_fee - eps
```

Comment:
I'm pretty skeptical of this. I'll circle back on this at the end of the review to double check on this.

Reply:
Added a comment to resolve later